crates/analyzer/src/traversal/expressions.rs

```diff
@@ -995,8 +995,139 @@ fn expr_call_path<T: std::fmt::Display>(
     generic_args: &Option<Node<Vec<fe::GenericArg>>>,
     args: &Node<Vec<Node<fe::CallArg>>>,
 ) -> Result<(ExpressionAttributes, CallType), FatalError> {
-    let named_thing = context.resolve_path(path, func.span)?;
-    expr_call_named_thing(context, named_thing, func, generic_args, args)
+    match context.resolve_any_path(path) {
```

Comment:
Why is `resolve_any_path` called instead of `resolve_visible_path`? If a "normal" associated function is not visible, it should not be counted as a callable candidate.

For example,  there is no ambiguity in the following example, but the compiler rejects it.
```rust
// a.fe
pub struct MyS {
    fn x() -> u256 {
        return 10
    }
}

// main.fe
use a::MyS

trait Trait {
    fn x() -> u256;
}

impl Trait for MyS {
    fn x() -> u256 {
        return 10
    }
}

fn main() -> u256 {
    return MyS::x()
}
```

Reply:
Good catch! Eagle eyes! :eagle: I pushed a solution in another commit. I'm not super happy with it but I'm not very creative this morning.